cmd/cmd.go

```diff
@@ -36,10 +38,9 @@ var (
 // 2. Create database (if schemaOnly is set to false)
 // 3. Run data conversion (if schemaOnly is set to false)
 // 4. Generate report
-func CommandLine(driver, targetDb, projectID, instanceID, dbName string, dataOnly, schemaOnly, skipForeignKeys bool, schemaSampleSize int64, sessionJSON string, ioHelper *conversion.IOStreams, outputFilePrefix string, now time.Time) error {
+func CommandLine(ctx context.Context, driver, targetDb, projectID, instanceID, dbName string, dataOnly, schemaOnly, skipForeignKeys bool, schemaSampleSize int64, sessionJSON string, ioHelper *conversion.IOStreams, outputFilePrefix string, now time.Time) error {
```

Comment:
I think we can go even further and pass dbURI to CommandLine() itself. The guiding principle is that dbURI is supposed to be a validated URI i.e., we should have already verified that projectID, instanceID are valid by the time CommandLine is called. The way we can make that work is by changing `conversion.AnalyzeError` to also take dbURI as the parameter.

That said, this PR already does a large amount of cleanup and we should get this submitted and move forward. The other cleanup that I have suggested can be easily done as a follow up.